### PR TITLE
contributing: add setup instructions for pkgs needed & building source

### PR DESCRIPTION
**Pull request by metral**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -5,6 +5,35 @@ Do you want to hack on Pulumi?  Awesome!  We are so happy to have you.
...
+1. Python: `python-setuptools`, `pip`, `pandoc`
+1. Go: [golangci-lint](https://github.com/golangci/golangci-lint)
+1. JS: `npm`, `yarn`
+
+### Building Source
+
+1. Setup `pulumi/scripts`:
+
+    ```
+    $ git clone https://github.com/pulumi/scripts ${GOPATH}/src/github.com/pulumi/scripts
```
Comment (hausdorff) [inline CONTRIBUTING.md]: Tiny, suggestion, how about: `$(go env GOPATH)`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -5,6 +5,35 @@ Do you want to hack on Pulumi?  Awesome!  We are so happy to have you.
...
+    ```
+
+1. Restore vendor dependencies:
+
+    ```
+    $ cd ${GOPATH}/src/github.com/pulumi/pulumi-kubernetes
+    $ ${GOPATH}/src/github.com/pulumi/scripts/ci/ensure-dependencies
```
[the remark this answers is not in the document]
Comment (metral) [reply CONTRIBUTING.md]: Yup, `make ensure` sufficed. Removing the dep on pulumi/scripts. Thanks!

**Comment by metral** [comment]
@hausdorff All feedback has been addressed. Please lmk if there's anything else missing.

**Comment by metral** [comment]
@hausdorff I've addressed your comments from Slack. PTAL

**Pull request merged**